subprojects/architecture-test/src/test/java/org/gradle/architecture/test/ArchUnitFixture.java

```diff
@@ -69,18 +69,26 @@
 import static com.tngtech.archunit.core.domain.JavaModifier.PUBLIC;
 import static com.tngtech.archunit.core.domain.properties.HasModifiers.Predicates.modifier;
 import static com.tngtech.archunit.core.domain.properties.HasName.Functions.GET_NAME;
+import static com.tngtech.archunit.core.domain.properties.HasName.Predicates.nameMatching;
 import static com.tngtech.archunit.core.domain.properties.HasType.Functions.GET_RAW_TYPE;
 import static java.util.stream.Collectors.toSet;
 
 public interface ArchUnitFixture {
-    DescribedPredicate<JavaMember> public_api_methods = declaredIn(gradlePublicApi())
-        .and(modifier(PUBLIC))
-        .as("public API methods");
-
     DescribedPredicate<JavaMember> not_written_in_kotlin = declaredIn(
             resideOutsideOfPackages("org.gradle.configurationcache..", "org.gradle.kotlin.."))
             .as("API written in Java");
 
+    DescribedPredicate<JavaMember> kotlin_internal_methods = declaredIn(gradlePublicApi())
+        .and(not(not_written_in_kotlin))
+        .and(modifier(PUBLIC))
+        .and(nameMatching(".+\\$[a-z_]+")) // Kotlin internal methods have `$kotlin_module_name` appended to their name
+        .as("Kotlin internal methods");
```

Comment:
🤔 To simplify this predicate, maybe only focus on one thing? Like the name matching? Since you are only using this condition negated, the result would be the same if you'd only have `nameMatching(...)` without all the other conditions, right?

Reply:
I could remove the `modifier(public)` predicate given the current usage site.
If it gets used elsewhere then it could create false positives on non-public members.
The rest would have to stay in order to not get false positives in the current usage site.
I'll leave it as is for now.